File: src/main/java/com/wanderers/hotelier_webservices/server/dao/api/BookingDao.java
package com.wanderers.hotelier_webservices.server.dao.api;

import com.wanderers.hotelier_webservices.server.dto.BookingDto;
import com.wanderers.hotelier_webservices.server.exception.BookingDaoException;

/**
 * Data access contract for the booking entity
 */
public interface BookingDao {

    /**
     * Create a booking record over the provided data
     *
     * @param booking
     * @return {@link BookingDto}
     * @throws BookingDaoException
     */
    BookingDto create(BookingDto booking) throws BookingDaoException;
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/dao/api/BookingDao.kt
package com.wanderers.hotelier_webservices.server.dao.api

import com.wanderers.hotelier_webservices.server.dto.BookingDto
import com.wanderers.hotelier_webservices.server.exception.BookingDaoException

/**
 * Data access contract for the booking entity
 */
interface BookingDao {
    /**
     * Create a booking record over the provided data
     *
     * @param booking
     * @return [BookingDto]
     * @throws BookingDaoException
     */
    @kotlin.Throws(BookingDaoException::class)
    fun create(booking: BookingDto?): BookingDto?
}